Add article: Simple Tailwind CSS 4 Setup for Jekyll (#235)
andycroll/andycroll.com: _posts/2026/2026-01-27-simple-tailwind-css-4-setup-for-jekyll.md

---
title: "Simple Tailwind CSS 4 Setup for Jekyll"
description: "A minimal setup for Tailwind CSS 4 in Jekyll using the jekyll-tailwind gem"
layout: article
category: ruby
date: 2026-01-27 09:00
image:
  base: "2026/simple-tailwind-css-4-setup-for-jekyll"
  alt: "Black and white airplane flying in the sky"
  credit: "Sander Weeteling"
  source: "https://unsplash.com/photos/iGDg_f_mlWo"
---

> Tailwind CSS 4 changed how configuration works. Gone is the JavaScript config file, replaced by CSS-based configuration using `@theme` directives. Here's how to set it up with Jekyll.

Author: ```suggestion
Tailwind CSS 4 changed how configuration works. The JavaScript config file has been replaced by CSS-based configuration using `@theme` directives and uses the `tailwind` CLI to shake down the generated tailwind classes and minify. Here's how to set it up with [Jekyll](https://jekyllrb.com).
```

## The Setup

> Three files, plus one more if you want plugins.

Author: ```suggestion
Changes to three files, plus one more if you want a couple of tailwind plugins.
```

### Gemfile

```ruby
gem "jekyll-tailwind", group: [:jekyll_plugins]
```

Run `bundle install` to fetch the gem. The [`jekyll-tailwind`](https://github.com/vormwald/jekyll-tailwind) gem handles everything. No separate build pipeline, no PostCSS config, no watching for changes. It hooks into Jekyll's build process.

> Under the hood, it uses [`tailwindcss-ruby`](https://github.com/flavorjones/tailwindcss-ruby)---the same gem that powers Tailwind in Rails.

Author: ```suggestion
Under the hood, it uses [`tailwindcss-ruby`](https://github.com/flavorjones/tailwindcss-ruby)---the same gem that powers the gems for Tailwind in Rails.
```
